pkg/api/handlers/events.go

```diff
@@ -27,15 +32,30 @@ func GetEvents(w http.ResponseWriter, r *http.Request) {
 		}
 	}
 
-	libpodEvents, err := getRuntime(r).GetEvents(libpodFilters)
-	if err != nil {
-		utils.BadRequest(w, "filters", strings.Join(r.URL.Query()["filters"], ", "), err)
+	if len(query.Since) > 0 || len(query.Until) > 0 {
+		fromStart = true
+	}
+	eventChannel := make(chan *events.Event)
+	go func() {
+		readOpts := events.ReadOptions{FromStart: fromStart, Stream: true, Filters: libpodFilters, EventChannel: eventChannel, Since: query.Since, Until: query.Until}
+		eventsError = getRuntime(r).Events(readOpts)
+	}()
+	if eventsError != nil {
+		utils.InternalServerError(w, eventsError)
 		return
 	}
-
-	var apiEvents = make([]*Event, len(libpodEvents))
-	for _, v := range libpodEvents {
-		apiEvents = append(apiEvents, EventToApiEvent(v))
+	w.Header().Set("Content-Type", "application/json")
+	w.WriteHeader(http.StatusOK)
+	for event := range eventChannel {
+		e := EventToApiEvent(event)
+		//utils.WriteJSON(w, http.StatusOK, e)
+		coder := json.NewEncoder(w)
+		coder.SetEscapeHTML(true)
+		if err := coder.Encode(e); err != nil {
+			logrus.Errorf("unable to write json: %q", err)
```

Comment:
I am hitting this.
```
ERRO[4557] unable to write json: "write unix /run/podman/podman.sock->@: i/o timeout"
```
Running `sudo podman --log-level=debug system service -t 0` for some time (When trying to reproduce a few minutes is enough), I can still communicate through the API just fine (so the socket is still active), but I was not getting events anymore and then noticed that on every event this message is printed.

Stopping `podman system service`, starting it back up and then connecting again from cockpit-podman works fine. But it should not timeout. (I am not quite sure what is timing out, the socket obviously not, maybe my connect, or this streamer).
Have you tried running it for longer and listening to the stream?

Reply:
Thanks for the report. I'll take a look.

Reply:
I can reproduce locally as well.

Reply:
This happens for other handlers (e.g., stats) as well, so it's unrelated to this PR.

Reply:
@baude @jwhonce, have you seen that before?